### ISSUE TITLE
Command `datahub version` yields "AttributeError: __name__" error

### ISSUE THREAD

**Issue by elvis-ds** [datahub-project/datahub#9760]
**Describe the bug**
When running the command `datahub version` the following error occurs:

```
Traceback (most recent call last):
  File "C:\Users\Anon\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Anon\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Anon\Documents\repos\datahub\.venv\Scripts\datahub.exe\__main__.py", line 4, in <module>
  File "C:\Users\Anon\Documents\repos\datahub\.venv\lib\site-packages\datahub\entrypoints.py", line 18, in <module>
    from datahub.cli.docker_cli import docker
  File "C:\Users\Anon\Documents\repos\datahub\.venv\lib\site-packages\datahub\cli\docker_cli.py", line 595, in <module>
    def quickstart(  # noqa: C901
  File "C:\Users\Anon\Documents\repos\datahub\.venv\lib\site-packages\click\decorators.py", line 373, in decorator
    _param_memo(f, cls(param_decls, **attrs))
  File "C:\Users\Anon\Documents\repos\datahub\.venv\lib\site-packages\click\core.py", line 2536, in __init__
    super().__init__(param_decls, type=type, multiple=multiple, **attrs)
  File "C:\Users\Anon\Documents\repos\datahub\.venv\lib\site-packages\click\core.py", line 2114, in __init__
    self.type: types.ParamType = types.convert_type(type, default)
  File "C:\Users\Anon\Documents\repos\datahub\.venv\lib\site-packages\click\types.py", line 1057, in convert_type
    return FuncParamType(ty)
  File "C:\Users\Anon\Documents\repos\datahub\.venv\lib\site-packages\click\types.py", line 167, in __init__
    self.name: str = func.__name__
  File "C:\Users\Anon\AppData\Local\Programs\Python\Python39\lib\typing.py", line 711, in __getattr__
    raise AttributeError(attr)
AttributeError: __name__
```

**To Reproduce**
1. Open a powershell terminal
2. Run the following commands:
```posh
python -m virtualenv .venv`
.venv/Scripts/activate.ps1`
python3 -m pip install --upgrade pip wheel setuptools
python3 -m pip install --upgrade acryl-datahub
datahub version
```

**Expected behavior**
I would expect the datahub version would be printed out like in other installations:
```
DataHub CLI version: 0.12.1.4
Python version: 3.9.13 (tags/...)
```

**Desktop (please complete the following information):**
 - Windows 10.0.19041.3636
 - Docker version 24.0.6, build ed223bc
 - Python 3.9.13

**Comment by ms2296** [comment]
Encountering the same error on a Mac, using conda for virtual environment. 
- MAC OS 14.1
- Conda Env with Python 3.9
- Docker v 24.0.7, build afdd53b

**Comment by hsheth2** [comment]
I'm able to repro this issue

**Comment by kzmain** [comment]
> Encountering the same error on a Mac, using conda for virtual environment.
> 
> * MAC OS 14.1
> * Conda Env with Python 3.9
> * Docker v 24.0.7, build afdd53b

I encountered the same issue on my Mac (M2 Pro Chip) today. The issue was solved after I upgraded the python version to 3.11
